docs/configuration-ini.md

```diff
@@ -53,227 +50,484 @@ Below is sample Network section.
-
-   Initial learning rate to decay
+2. ```beta1 = <float>```
 
-3. ```beta1 = <float>```
+   beta1 parameter for adam optimizer. Only valid for adam. The default value is 0.9.
 
-   beta1 parameter for adam optimizer. Only valid for adam.   0.9 is default.
 
-4. ```beta2 = <float>```
+3. ```beta2 = <float>```
 
-   beta2 parameter for adam optimizer. Only valid for adam. 0.999 is default.
+   beta2 parameter for adam optimizer. Only valid for adam. The default value is 0.999.
 
-5. ```epsilon = <float>```
+4. ```epsilon = <float>```
 
-     Epsilon parameter for adam optimizer. Only valid for adam. 1.0e-7 is default.
+   Epsilon parameter for adam optimizer. Only valid for adam. The default value is 1.0e-7.
 
 Below is a sample Optimizer section.
 
 ```ini
 # Optimizer Section
 [Optimizer]
-Type = adam
-Learning_rate = 1e-4
-batch_size = 32
+type = adam
 beta1 = 0.9
 beta2 = 0.999
 epsilon = 1e-7
 ```
 
-### DataSet Section
-
-Define data set. training, validation, test data set.
-
-Start with "[ DataSet ]"
+### Learning Rate Scheduler Section
 
-1. ```buffersize = <unsigned int>```
+Define the type, learning rate, decay steps and decay rate.
 
-    Define Buffer size. usually it is greater than batch size.
+Start with "[LearningRateScheduler]"
 
-    Data buffer thread keeps reading the data from the file and stores the data into the data buffer.
-    Meanwhile main thread gets the training data from this data buffer and feeds it to the model.
-    This keyword defines the size of Data Buffer.
-
-2. ```traindata = <string>```
-
-    training data file path.   The data must be saved as following
+1. ```type = <string>```
 
-    ```feature data[i], label data[i], feature data[i+1], label data[i+1], ...```
+   constant, exponential and step are supported.
+      * constant : constant learning rate
+      * exponential : exponential decay
+      * step: step decay
 
-3. ```validdata = <string>```
+2. ```learning_rate = <float>```
 
-    validation data file path.   The data must be saved as following
+   Initial learning rate to decay.
 
-    ```feature data[i], label data[i], feature data[i+1], label data[i+1], ...```
+   Constant and exponential receive only one float value.
 
-4. ```testdata = <string>```
+   However, Step must receive two or more float values seperated by commas.
 
-    test data file path.   The data must be saved as following
+   ```learning_rate = <float>, <float>, ..., <float>```
 
-    ```feature data[i], label data[i], feature data[i+1], label data[i+1], ...```
 
-5. ```labeldata = <string>```
+3. ```decay_steps = <float>```
 
-    label data file path. The data must be saved as following
+   Decay steps. Only valid for exponential.
 
-    ```class Name [i], class name [i+1],...```
+4. ```decay_rate = <float>```
 
-### Layer Section
+   Decay rate
 
-Describe hyper-parameters for layer. Order of layers in the model follows the order of definition of layers here from top to bottom.
+5. ```iteration = <unsigned int>```
```

Comment:
iteration could also be a vector of unsigned int

Reply:
okay